src/Misc.ino

```diff
@@ -1581,41 +1581,173 @@ String parseTemplate(String &tmpString, byte lineSize)
+                                    break;
+                                case 'D' ://Dx.y min 'x' digits zero filled & 'y' decimal fixed digits
+                                  int x;
+                                  int y;
+                                  x = 0;
+                                  y = 0;
+
+                                  switch (tempValueFormat.length())
+                                  {
+                                    case 2: //Dx
+                                      if (tempValueFormat[1]>='0' && tempValueFormat[1]<='9')
+                                      {
+                                        x = (int)tempValueFormat[1]-'0';
+                                      }
+                                      break;
+                                    case 3: //D.y
+                                      if (tempValueFormat[1]=='.' && tempValueFormat[2]>='0' && tempValueFormat[2]<='9')
+                                      {
+                                        y = (int)tempValueFormat[2]-'0';
+                                      }
+                                      break;
+                                    case 4: //Dx.y
+                                      if (tempValueFormat[1]>='0' && tempValueFormat[1]<='9' && tempValueFormat[2]=='.' && tempValueFormat[3]>='0' && tempValueFormat[3]<='9')
+                                      {
+                                        x = (int)tempValueFormat[1]-'0';
+                                        y = (int)tempValueFormat[3]-'0';
+                                      }
+                                      break;
+                                    case 1: //D
+                                    default: //any other combination x=0; y=0;
+                                      break;
+                                  }
+                                  value = toString(valFloat,y);
+                                  int indexDot;
+                                  indexDot = value.indexOf('.') > 0 ? value.indexOf('.') : value.length();
+                                  for (byte f = 0; f < x - indexDot; f++)
+                                    value = "0" + value;
+                                  break;
+                                case 'F' :// FLOOR (round down)
+                                  value = (int)floorf(valFloat);
+                                  break;
+                                case 'E' :// CEILING (round up)
+                                  value = (int)ceilf(valFloat);
+                                  break;
+                                default:
+                                  value = "ERR";
+                                  break;
+                                }
+
+                                // Check Justification syntax
+                                if (valueJust.length() > 0) //do the checks only if a Justification is defined to optimize loop
```

Comment:
When doing optimizations, better store the length in a const variable :)
It is being used in the lines below.